src/Microsoft.FeatureManagement/Targeting/TargetingEvaluator.cs

```diff
@@ -0,0 +1,368 @@
+            {
+                throw new ArgumentException(reason, paramName);
+            }
+
+            if (settings.Audience.Exclusion != null)
+            {
+                //
+                // Check if the user is in the exclusion directly
+                if (targetingContext.UserId != null &&
+                    settings.Audience.Exclusion.Users != null &&
+                    settings.Audience.Exclusion.Users.Any(user => targetingContext.UserId.Equals(user, GetComparisonType(ignoreCase))))
+                {
+                    return false;
+                }
+
+                //
+                // Check if the user is in a group within exclusion
+                if (targetingContext.Groups != null &&
+                    settings.Audience.Exclusion.Groups != null &&
+                    settings.Audience.Exclusion.Groups.Any(group => targetingContext.Groups.Any(g => g?.Equals(group, GetComparisonType(ignoreCase)) ?? false)))
+                {
+                    return false;
+                }
+            }
+
+            //
+            // Check if the user is being targeted directly
+            if (settings.Audience.Users != null &&
+                IsTargeted(
+                    targetingContext.UserId,
+                    settings.Audience.Users,
+                    ignoreCase))
+            {
+                return true;
+            }
+
+            //
+            // Check if the user is in a group that is being targeted
+            if (settings.Audience.Groups != null &&
+                IsTargeted(
+                    targetingContext,
+                    settings.Audience.Groups,
+                    ignoreCase,
+                    hint))
+            {
+                return true;
+            }
+
+            //
+            // Check if the user is being targeted by a default rollout percentage
+            return IsTargeted(
+                targetingContext,
+                settings.Audience.DefaultRolloutPercentage,
+                ignoreCase,
+                hint);
+        }
+
+        /// <summary>
+        /// Determines if a targeting context is targeted by presence in a list of users
+        /// </summary>
+        public static bool IsTargeted(
+            string userId,
+            IEnumerable<string> users,
+            bool ignoreCase)
+        {
+            if (users == null)
+            {
+                throw new ArgumentNullException(nameof(users));
+            }
+
+            if (userId != null &&
+                users.Any(user => userId.Equals(user, GetComparisonType(ignoreCase))))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Determines if targeting context is targeted by presence in a list of groups
+        /// </summary>
+        public static bool IsTargeted(
+            IEnumerable<string> targetingContextGroups,
```

Comment:
Instead of `targetingContextGroups` I'd suggest source groups.

And instead of `groups`, I'd suggest `targetedGroups`